### PR TITLE
refactor!: remove support for state from operation

### PR DESCRIPTION
**Pull request by iamsauravsharma**
- remove support of state from operation and directly use struct fields if required for passing state. currently state can be passed in two ways which make them more confusing about where to pass them

### CODE REVIEW

File: examples/postgres/migrations/m0004_complex_operation.rs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            .execute(connection)
+            .await?;
+        Ok(())
+    }
+}
+
+pub(crate) struct M0004Migration {
+    pub(crate) id: i32,
+    pub(crate) message: String,
+}
```
Comment (SimenB) [inline examples/postgres/migrations/m0004_complex_operation.rs:34]: Hey! 👋 

with this new approach, how would a `m0005` migration refer to this migration from `parents`? Would I then have to pass through whatever state one of the migrations needs to every single subsequent migration?

Comment (iamsauravsharma) [reply examples/postgres/migrations/m0004_complex_operation.rs:34]: You can use virtual migration, For example for above you can use `("main", "m0004_complex_operation")`

Comment (iamsauravsharma) [reply examples/postgres/migrations/m0004_complex_operation.rs:34]:
```
fn parents(&self) -> Vec<Box<dyn Migration<Postgres>>> {
    vec![("main", "m0004_complex_operation")]
}
 ```

Comment (iamsauravsharma) [reply examples/postgres/migrations/m0004_complex_operation.rs:34]: I have added example https://github.com/iamsauravsharma/sqlx_migrator/blob/master/examples/postgres/migrations/m0005_reference_complex.rs

**Pull request merged**